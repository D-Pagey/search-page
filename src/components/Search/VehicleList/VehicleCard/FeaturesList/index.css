.features-container {
  align-self: center;
  margin-top: 3px;
  width: 266px;
}

.features-list {
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
}

.feature {
  color: #7d807c;
  font-size: 12px;
  padding-left: 5px;
}