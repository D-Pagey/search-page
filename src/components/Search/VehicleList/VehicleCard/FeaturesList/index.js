import React from 'react';
import styled from 'styled-components';
import { arrayOf, string } from 'prop-types';

import titleCase from '../../../../../utils/capitalize';

const FeaturesListWrapper = styled.div`
  align-self: center;
  margin-top: 3px;
  @media (min-width: 768px) {
    width: 100%;
  }
`;

const Features = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0 10px;
`;

const FeatureItem = styled.li`
  color: #7d807c;
  font-size: 12px;
  padding-left: 5px;
`;

function Feature({ feature }) {
  const formatted = titleCase(feature.replace(/_/g, ' '));

  return (
    <FeatureItem>{formatted} â€¢</FeatureItem>
  );
}

export default function FeaturesList({ features }) {
  return (
    <FeaturesListWrapper>
      <Features>
        {features.map(element => <Feature feature={element} key={element} />)}
      </Features>
    </FeaturesListWrapper>
  );
}

Feature.propTypes = {
  feature: string.isRequired,
};

FeaturesList.propTypes = {
  features: arrayOf(string).isRequired,
};
