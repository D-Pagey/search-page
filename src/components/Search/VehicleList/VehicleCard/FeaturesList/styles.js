import styled from 'styled-components';

export const FeaturesListWrapper = styled.div`
  align-self: center;
  margin-top: 3px;
  @media (min-width: 768px) {
    width: 100%;
  }
`;

export const Features = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0 10px;
`;

export const FeatureItem = styled.li`
  color: #7d807c;
  font-size: 12px;
  padding-left: 5px;
`;
