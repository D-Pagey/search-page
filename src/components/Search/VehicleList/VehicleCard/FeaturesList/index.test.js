import React from 'react';
import { shallow } from 'enzyme';
import FeaturesList from '.';

it('should render', () => {
  const wrapper = shallow(<FeaturesList features={[]} />);
  expect(wrapper).toMatchSnapshot();
});
