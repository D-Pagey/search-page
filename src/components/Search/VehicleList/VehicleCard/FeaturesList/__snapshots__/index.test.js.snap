// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="features-container"
>
  <ul
    className="features-list"
  />
</div>
`;
